src/xcode/ENA/ENA/Source/Services/Event Registration/TraceWarningDownload/TraceWarningPackageDownload.swift

```diff
@@ -0,0 +1,400 @@
+	func startTraceWarningPackageDownload(
+		with appConfiguration: SAP_Internal_V2_ApplicationConfigurationIOS,
+		completion: @escaping (Result<TraceWarningSuccess, TraceWarningError>) -> Void
+	)
+}
+
+enum TraceWarningDownloadStatus {
+	case idle
+	case checkingForNewPackages
+	case downloading
+}
+
+class TraceWarningPackageDownload: TraceWarningPackageDownloading {
+	
+	// MARK: - Init
+	
+	init(
+		client: Client,
+		store: Store,
+		eventStore: EventStoringProviding,
+		countries: [Country.ID] = ["DE"],
+		signatureVerifier: SignatureVerification = SignatureVerifier()
+	) {
+		self.client = client
+		self.store = store
+		self.eventStore = eventStore
+		self.countries = countries
+		self.signatureVerifier = signatureVerifier
+		
+		self.matcher = TraceWarningMatcher(eventStore: eventStore)
+	}
+	
+	// MARK: - Overrides
+	
+	// MARK: - Protocol TraceWarningPackageDownloading
+	
+	var statusDidChange: ((TraceWarningDownloadStatus) -> Void)?
+	
+	func startTraceWarningPackageDownload(
+		with appConfiguration: SAP_Internal_V2_ApplicationConfigurationIOS,
+		completion: @escaping (Result<TraceWarningSuccess, TraceWarningError>) -> Void
+	) {
+		Log.info("TraceWarningPackageDownload: Start was triggered.", log: .checkin)
+		
+		// Cancel Download if a download is still in progress
+		guard status == .idle else {
+			Log.info("TraceWarningPackageDownload: Aborted due to already running download.", log: .checkin)
+			completion(.failure(.downloadIsRunning))
+			return
+		}
+		
+		// 1. Check for check-ins
+		guard !eventStore.checkinsPublisher.value.isEmpty else {
+			// If the database is empty, remove all traceWarningPackageMetadatas
+			let packagesToDelete = eventStore.traceWarningPackageMetadatasPublisher.value
+			removePackagesFromTraceWarningMetadataPackagesTable(packagesToDelete)
+			Log.info("TraceWarningPackageDownload: Aborted due to checkin database is empty.", log: .checkin)
+			completion(.success(.noCheckins))
+			return
+		}
+
+		status = .checkingForNewPackages
+		
+		checkForDownloadTraceWarningPackages(with: appConfiguration, countries: countries, completion: { [weak self] result in
+			self?.status = .idle
+			
+			switch result {
+			
+			case let .success(success):
+				Log.info("TraceWarningPackageDownload: Completed processing packages!", log: .checkin)
+				completion(.success(success))
+			case let .failure(error):
+				Log.info("TraceWarningPackageDownload: Failed processing packages with error: \(error)")
+				completion(.failure(error))
+			}
+		})
+	}
+	
+	// MARK: - Public
+	
+	// MARK: - Internal
+	
+	/// Determines the earliest relevant package == the min of the checkin database table. Not private for testing purposes.
+	var earliestRelevantPackage: Int? {
+		let minCheckin = eventStore.checkinsPublisher.value.min(by: { $0.checkinStartDate < $1.checkinStartDate })
```

Comment:
The TechSpec says StartDate. 
This makes sense because we want here to filter out the earliest package. Earliest is in that case the startDate. Everything that is older then the earliest startDate is not relevant.